bootable-jar/WFLY-13305_Bootable_Jar.adoc

```diff
@@ -0,0 +1,505 @@
+This is similar to the _--properties=<properties file>_ option that you can provide to JBoss CLI command line. 
+This http://www.mastertheboss.com/jboss-server/jboss-script/using-properties-in-cli-scripts[article] covers the feature.
+
+====== Configuring CLI scripts execution
+
+CLI script files are text files that contain a sequence of WildFly CLI commands. Commands can be CLI defined commands 
+(some builtin commands allowing to achieve complex sequence of server operations) and generic management operations to be sent to the server. Some examples can
+be found in WildFly administration guide https://docs.wildfly.org/20/Admin_Guide.html#CLI_Recipes[CLI recipes chapter].
+
+In the context of Bootable JAR, the script does not need to contain commands to connect to the server or start an embedded server. 
+The Maven plugin handles that for you by starting an embedded server for each group of scripts.
+
+The plugin allows you to execute multiple groups of scripts with different CLI contexts. 
+A group of scripts and its configuration are defined in a ```cli-session``` composed of:
+
+* ```<script-files>```: the list of paths to script files .
+* ```properties-file```: (optional) a path to a properties file that contains java properties that scripts can reference (using the syntax ```${my.prop}```). 
+For example, a command that sets the public inet-address to the value of ```all.addresses``` system property looks like: ```/interface=public:write-attribute(name=inet-address,value=${all.addresses})```
+* ```resolve-expressions```: (optional) a boolean indicating if system properties or 
+expressions are resolved before sending the operation requests to the server. Value is ```true``` by default.
+
+All scripts present in a ```cli-session``` are executed within a single CLI execution. An embedded server is started for each defined ```cli-session```.
+
+NB: The scripts are executed in the order they are defined in the plugin configuration. 
+
+CLI configuration example:
+```
+<cli-sessions>
+  <cli-session>
+    <script-files>
+        <script>../scripts/script1.cli</script>
+    </script-files>
+    <!-- We want the env variables to be resolved during server execution -->
+    <resolve-expressions>false</resolve-expressions>
+  </cli-session>
+  <cli-session>
+    <script-files>
+        <script>../scripts/script2.cli</script>
+    </script-files>
+    <properties-file>../scripts/cli.properties</properties-file>
+    <!-- We want the properties to be resolved during CLI execution (not actually needed, this is the default behavior) -->
+    <resolve-expressions>true</resolve-expressions>
+  </cli-session>
+</cli-sessions>
+```
+
+===== Logging configuration
+
+During packaging the maven plugin generates a logging.properties file that reflects the server logging configuration. This ensures that the bootable JAR 
+will boot with a logging configuration in sync with the logging subsystem.
```

Comment:
What default configuration is given to the `server logging` subsystem? 
What does it mean that the `logging.properties` file is reflecting that configuration?

Reply:
Some context: when server starts, in order to have JBoss logging enabled and configured before the logging subsystem is initialized, WildFly relies on a logging.properties file that configures JBoss logging.
WildFly distribution contains a default logging.properties.
In a bootable JAR context, we could have CLI scripts that tune the logging subsystem. So at boot, if the logging.properties file is the WildFly distribution default one, the logging subsystem changes would be not taken into account before the logging subsystem is initialized. By generating a logging.properties from the logging subsystem, we capture the logging subsystem configuration. JBoss logging uses this file so is configured in an identical way as the logging subsystem.

The logging subsystem defaults come from the "logging" Galleon layer (that contains a configuration identical to WildFly default xml configurations).
If no logging subsystem is present in the server configuration (so no Galleon logging layer provisioned), we rely on the default loging.properties file.

Reply:
Thanks @jfdenise - this is quite a hack, right? :) Let0s wait for the final version of the subsystem and then generate the logging.properties out of it.
Sorry but without the example involving custom CLI configuration I don't think this is easily understandable.
I'd appreciate if such a reasoning could be put into the proposal, as well.

Reply:
Added more details.